lib/protractor.js

```diff
@@ -700,9 +696,33 @@ Protractor.prototype.initDebugger_ = function(debuggerClientPath, opt_debugPort)
       return found;
     });
   };
+  for (var key in global) {
+    context[key] = global[key];
+  }
   var sandbox = vm_.createContext(context);
-  
+
   var browserUnderDebug = this;
+  flow.execute(function() {    
+    log.puts('Starting WebDriver debugger in a child process. Pause is ' +
+        'still beta, please report issues at github.com/angular/protractor\n');
+    process.debugPort = opt_debugPort || process.debugPort;
+    onPortAvailable(process.debugPort, function() {
+      var args = [process.pid, process.debugPort];
+      if (browserUnderDebug.debuggerServerPort_) {
+        args.push(browserUnderDebug.debuggerServerPort_);
+      }
+      var nodedebug = require('child_process').fork(debuggerClientPath, args);
+      process.on('exit', function() {
+        nodedebug.kill('SIGTERM');
+      }); 
+    });
+  });
+  
+  var pausePromise = flow.timeout(1000, 'waiting for debugger to attach')
+      .then(function() {
+        // Necessary for backward compatibility with node < 0.12.0
+        browserUnderDebug.executeScript_('', 'empty debugger hook');
```

Comment:
It's because the way node debugger changed in 0.12.
Before, when you start the debug process, it starts a server but the program continues (that's why you needed the sleep() in there to manually stop execution while waiting for the debugger to attach). However, now, the moment you start debug, the program stops and waits for the reqContinue signal. 

As a result of that, before, we call debug in this process. In the client, we take control once we attach.
Now, we call debug from the client. Because the program is paused, reqContinue is the first call we make. After that, we can take control as before. 

However, if we're using node 0.10, the program isn't paused when issued the debug, so the first reqContinue actually makes the program jump to the next break point.